src/modules/auth/auth.schema.ts

```diff
@@ -1,70 +1,135 @@
-import { z } from 'zod';
-import { buildJsonSchemas } from 'fastify-zod';
+import { FromSchema } from 'json-schema-to-ts';
 
-const userCore = {
-	name: z.string().min(1),
-	email: z
-		.string({
-			required_error: 'Email is required',
-			invalid_type_error: 'Email must be a string',
-		})
-		.min(1)
-		.email(),
-};
-
-const createUserSchema = z.object({
-	...userCore,
-	password: z
-		.string({
-			required_error: 'Password is required',
-			invalid_type_error: 'Password must be a string',
-		})
-		.min(8),
-});
-
-const createUserResponseSchema = z.object({
-	...userCore,
-});
+const createUserSchema = {
+	$id: 'createUserSchema',
+	type: 'object',
+	properties: {
+		name: {
+			type: 'string',
+			minLength: 1,
+			errorMessage: {
+				type: 'name.type',
+				minLength: 'name.minLength',
+			},
+		},
+		email: {
+			type: 'string',
+			format: 'email',
+			errorMessage: {
+				type: 'email.type',
+				format: 'email.format',
+			},
+		},
+		password: {
+			type: 'string',
+			minLength: 8,
+			errorMessage: {
+				type: 'password.type',
+				minLength: 'password.minLength',
+			},
+		},
+	},
+	required: ['email', 'password', 'name'],
+	errorMessage: {
+		required: {
+			email: 'email.required',
+			password: 'password.required',
+			name: 'name.required',
+		},
+	},
+} as const;
```

Comment:
Why as const?

Reply:
FromSchema<> expects the type of a const - Dosen't impact the code

https://github.com/ThomasAribart/json-schema-to-ts

Reply:
🤡